SCons/Tool/ninja/__init__.py

```diff
@@ -419,7 +418,7 @@ def ninja_execute(self):
         target = self.targets[0]
         if target.get_env().get('NINJA_SKIP'):
             return
-        if target.check_attributes('ninja_file') is None or not target.is_conftest:
+        if target.check_attributes('ninja_file') is None and not target.is_conftest():
```

Comment:
I'm guessing we don't have a test which caught that this was wrong, so we should probably add one? or enhance an existing test to catch this?

Reply:
okay I'll add a ninja conf test

Reply:
adding a test made me realize that the is_conftest check here is not needed. This check is inside the taskmaster.Task execute function, which for SConf it has its own execute, so ninja never sees those task executes. Its only needed in the builder execute.

The test I wrote creates a failing conftest and makes sure the ninja generation and build still proceeds, but it is dependent on is_conftest working so it will need to based off https://github.com/SCons/scons/pull/3980